lib/main.dart

```diff
@@ -128,125 +112,50 @@ class _MyAppState extends State<MyApp> {
-      headers: <String, String>{'Authorization': 'Bearer $accessToken'},
-    );
-
-    if (response.statusCode == 200) {
-      return jsonDecode(response.body);
-    } else {
-      throw Exception('Failed to get user details');
-    }
-  }
-
   Future<void> loginAction() async {
     setState(() {
       isBusy = true;
       errorMessage = '';
     });
 
     try {
-      final AuthorizationTokenResponse result =
-          await appAuth.authorizeAndExchangeCode(
-        AuthorizationTokenRequest(
-          AUTH0_CLIENT_ID,
-          AUTH0_REDIRECT_URI,
-          issuer: 'https://$AUTH0_DOMAIN',
-          scopes: <String>['openid', 'profile', 'offline_access'],
-          // promptValues: ['login']
-        ),
-      );
-
-      final Map<String, Object> idToken = parseIdToken(result.idToken);
-      final Map<String, Object> profile =
-          await getUserDetails(result.accessToken);
-
-      await secureStorage.write(
-          key: 'refresh_token', value: result.refreshToken);
+      final credentials = await auth0.webAuthentication(scheme: 'flutterdemo').login();
 
       setState(() {
         isBusy = false;
-        isLoggedIn = true;
-        name = idToken['name'];
-        picture = profile['picture'];
+        _credentials = credentials;
       });
     } on Exception catch (e, s) {
       debugPrint('login error: $e - stack: $s');
 
       setState(() {
         isBusy = false;
-        isLoggedIn = false;
         errorMessage = e.toString();
       });
     }
   }
 
   Future<void> logoutAction() async {
-    await secureStorage.delete(key: 'refresh_token');
+    await auth0.webAuthentication(scheme: 'flutterdemo').logout();
+
     setState(() {
-      isLoggedIn = false;
-      isBusy = false;
+      _credentials = null;
     });
   }
 
   @override
   void initState() {
-    initAction();
     super.initState();
-  }
 
-  Future<void> initAction() async {
-    final String storedRefreshToken =
-        await secureStorage.read(key: 'refresh_token');
-    if (storedRefreshToken == null) return;
-
-    setState(() {
-      isBusy = true;
-    });
-
-    try {
-      final TokenResponse response = await appAuth.token(TokenRequest(
-        AUTH0_CLIENT_ID,
-        AUTH0_REDIRECT_URI,
-        issuer: AUTH0_ISSUER,
-        refreshToken: storedRefreshToken,
-      ));
-
-      final Map<String, Object> idToken = parseIdToken(response.idToken);
-      final Map<String, Object> profile =
-          await getUserDetails(response.accessToken);
-
-      await secureStorage.write(
-          key: 'refresh_token', value: response.refreshToken);
-
-      setState(() {
-        isBusy = false;
-        isLoggedIn = true;
-        name = idToken['name'];
-        picture = profile['picture'];
-      });
-    } on Exception catch (e, s) {
-      debugPrint('error on refresh token: $e - stack: $s');
-      await logoutAction();
-    }
+    auth0 = Auth0('dev-bajcmartinez.eu.auth0.com', '4OZnEzQiVE8t1YguYMKsVReXDdHiH5Cj');
```

Comment:
Is it necessary to have an actual tenant domain and client id here, instead of placeholders for the developer to replace?

Reply:
No, it is not at all necessary, I'm replacing it with placeholders. Thanks for the catch!